### DISCUSSION TITLE
Help: how to comment lua code in vimscript?

### DISCUSSION THREAD

**Discussion by krisfans** [numToStr/Comment.nvim#289] [Q&A] [answered]
for example ,there is a piece of code in my vimrc, what should I do to comment `bar1` with `Comment.nvim`?
vimrc
```vim
foo
lua << EOF
bar1
bar2
EOF
```
I know [caw.vim](https://github.com/tyru/caw.vim) together with [context_filetype.vim](https://github.com/Shougo/context_filetype.vim) can deal with this situation, may be helpful.
Thanks.

**Comment by numToStr** [comment, marked as answer]
You just need vim and lua tree-sitter parsers installed and everything should work just fine.

**Reply by krisfans** [reply]
Thanks, It works. 
In addtion, Can Comment.nvim comment vimscript or lua surrounded by toml language? In this situation, some hooks in `.toml` such as `hook_add` will indicate that the next code's filetype is `vimrc` and [context_filetype.vim](https://github.com/Shougo/context_filetype.vim/blob/e276626e441eee2c624b9192113f1484bc2bc0f3/autoload/context_filetype/defaults.vim#L293) plugin has made a patch for this situation.

**Reply by numToStr** [reply]
It'll work, if proper injections are in place. Rule of thumb, if neovim can highlight it as such, then it will always work with Comment.nvim.
